### PR TITLE
CNV-13187: Create fedora-realtime container image

### PR DESCRIPTION
**Pull request by jordigilh**
The purpose of this PR is to add the job that builds the fedora-realtime-container-disk image configured to run with the real time kernel and tune packages. This image is to be used as part of the functional tests to validate the VMI realtime knobs implemented in this [PR](https://github.com/kubevirt/kubevirt/pull/6182) to add support for low latency workloads that run in real time.

This PR depends on this other [PR](https://github.com/kubevirt/kubevirtci/pull/665) from kubevirtci that contains the scripts to build the VM at runtime.

In short order, there are 3 PRs aimed at improving the support for realtime workloads by exposing new knobs in the VM schema targeted at reducing the CPU latency in the running workload:
* Kubevirt: https://github.com/kubevirt/kubevirt/pull/6182
* KubevirtCI: https://github.com/kubevirt/kubevirtci/pull/665
* project-infra: This one.

Please review @dhiller

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml [hunk 1]
```diff
@@ -57,3 +57,33 @@ postsubmits:
           resources:
             requests:
               memory: "8Gi"
+
+    - name: publish-centos8-realtime-container-image
+      always_run: false
```
Comment (dhiller) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml:62]: To fix [this check](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_project-infra/1515/pull-project-infra-check-testgrid-config/1430170427614826496) you should add the annotation that excludes this job from testgrid

```suggestion
      annotations:
        testgrid-create-test-group: "false"
      always_run: false
```

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml [hunk 2]
```diff
@@ -57,3 +57,33 @@ postsubmits:
...
+      run_if_changed: "cluster-provisioning/images/centos8-realtime/.*"
+      decorate: true
+      labels:
+        preset-dind-enabled: "true"
+        preset-docker-mirror-proxy: "true"
+        preset-kubevirtci-quay-credential: "true"        
+      cluster: phx-prow
```
Comment (dhiller) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml:69]:
```suggestion
      cluster: prow-workloads
```

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml [hunk 3]
```diff
@@ -260,3 +260,30 @@ presubmits:
...
       nodeSelector:
         type: bare-metal-external
+  - name: build-centos8-realtime-container-image
+    always_run: false
+    run_if_changed: "cluster-provisioning/images/centos8-realtime/.*"
+    decorate: true
+    labels:
+      preset-dind-enabled: "true"
+      preset-docker-mirror-proxy: "true"
+    cluster: phx-prow
```
Comment (dhiller) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml:270]:
```suggestion
      cluster: prow-workloads
```

**Review by dhiller — APPROVED** [review]
/approve

**Comment by dhiller** [comment]
Direction looks fine, please fix the small issues I found.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml [hunk 4]
```diff
@@ -57,3 +57,34 @@ postsubmits:
...
+      run_if_changed: "cluster-provisioning/images/vm-image-builder/fedora-realtime/.*"
+      decorate: true
+      labels:
+        preset-dind-enabled: "true"
+        preset-docker-mirror-proxy: "true"
+        preset-kubevirtci-quay-credential: "true"        
+      cluster: prow-workloads
+      spec:
+        containers:
+          - image: quay.io/kubevirtci/golang:v20210316-d295087
```
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml:74]: Why have you selected the golang image? As I understand it you are going to build and publish container image, right? I think `quay.io/kubevirtci/bootstrap` would be better for that.

**Review by fgimenez — COMMENTED** [review]
> @dhiller @fgimenez Do we really need to have this job? I noticed that there is no equivalent job for the `fedora-with-test-tooling`.

I think so, if we don't have something similar for `fedora-with-test-tooling` we should create it, I'll take care.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml (hunk 4 above)
Comment (jordigilh) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml:74]: Changed to `quay.io/kubevirtci/bootstrap:v20210316-d295087`. Confirmed that the image and tag exists :smile:

Comment (fgimenez) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml:74]: Looks like you didn't push the changes? I still see the golang image in the diff

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml [hunk 5]
```diff
@@ -57,3 +57,34 @@ postsubmits:
           resources:
             requests:
               memory: "8Gi"
+
+    - name: publish-fedora-realtime-container-image
+      annotations:
+        testgrid-create-test-group: "false"
+      always_run: false
+      run_if_changed: "cluster-provisioning/images/vm-image-builder/fedora-realtime/.*"
```
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml:65]:
```suggestion
      run_if_changed: "cluster-provision/images/vm-image-builder/fedora-realtime/.*"
```

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml [hunk 6]
```diff
@@ -57,3 +57,34 @@ postsubmits:
...
+        containers:
+          - image: quay.io/kubevirtci/golang:v20210316-d295087
+            command:
+              - "/usr/local/bin/runner.sh"
+              - "/bin/bash"
+              - "-ce"
+              - |
+                cat $QUAY_PASSWORD | docker login --username $(<$QUAY_USER) --password-stdin quay.io
+                ./cluster-provisioning/images/vm-image-builder/create-containerdisk.sh fedora-realtime
+                ./cluster-provisioning/images/vm-image-builder/publish-containerdisk.sh fedora-realtime quay.io/kubevirt/fedora-realtime-container-disk:latest
```
Comment (fgimenez) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml:82]:
```suggestion
                ./cluster-provision/images/vm-image-builder/create-containerdisk.sh fedora-realtime
                ./cluster-provision/images/vm-image-builder/publish-containerdisk.sh fedora-realtime quay.io/kubevirt/fedora-realtime-container-disk:latest
```

**Review by fgimenez — COMMENTED** [review]
I've triggered the job with the changes in the kubevirtci PR and it failed https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/publish-fedora-realtime-container-image/1442753980756135936#1:build-log.txt%3A76
```
/home/prow/go/src/github.com/jordigilh/kubevirtci/cluster-provision/images/vm-image-builder/customize-image.sh: line 25: cloud-localds: command not found
```
and later at https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/publish-fedora-realtime-container-image/1442753980756135936#1:build-log.txt%3A82
```
/home/prow/go/src/github.com/jordigilh/kubevirtci/cluster-provision/images/vm-image-builder/customize-image.sh: line 11: virsh: command not found
```
We probably need to make sure that the prerequisites are installed https://github.com/jordigilh/kubevirtci/tree/cnv_13187/create_centos8_stream_image/cluster-provision/images/vm-image-builder#prerequisites

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml (hunk 4 above)
Comment (jordigilh) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml:74]: Looks like I changed the wrong file. I'll push the correct changes. Thanks!

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml (hunk 6 above)
Comment (jordigilh) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml:82]: Running the command locally to confirm that it does not have any more typo(s)... :smile:

Comment (jordigilh) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml]: Command successful. I have pushed the correct changes now.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml (hunk 5 above)
Comment (jordigilh) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-postsubmits.yaml]: Fixed

**Comment by jordigilh** [comment]
> I've triggered the job with the changes in the kubevirtci PR and it failed https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/publish-fedora-realtime-container-image/1442753980756135936#1:build-log.txt%3A76
> 
> ```
> /home/prow/go/src/github.com/jordigilh/kubevirtci/cluster-provision/images/vm-image-builder/customize-image.sh: line 25: cloud-localds: command not found
> ```
> 
> and later at https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/publish-fedora-realtime-container-image/1442753980756135936#1:build-log.txt%3A82
> 
> ```
> /home/prow/go/src/github.com/jordigilh/kubevirtci/cluster-provision/images/vm-image-builder/customize-image.sh: line 11: virsh: command not found
> ```
> 
> We probably need to make sure that the prerequisites are installed https://github.com/jordigilh/kubevirtci/tree/cnv_13187/create_centos8_stream_image/cluster-provision/images/vm-image-builder#prerequisites

Weird, should not be using `vm-image-builder`. Can you try again with the current code?

**Comment by fgimenez** [comment]
> Weird, should not be using `vm-image-builder`. Can you try again with the current code?

In any case we will need the dependencies mentioned here https://github.com/kubevirt/kubevirtci/tree/main/cluster-provision/images/vm-image-builder#prerequisites The image added here should work for that, using `quay.io/kubevirtci/vm-image-builder:v20210928-7fbacd6` we just need to start libvirtd and virtlogd before calling the build script:
```
/usr/sbin/libvirtd > /var/log/libvirt/libvirtd.log 2>&1 &
/usr/sbin/virtlogd > /var/log/libvirt/virtlogd.log 2>&1 &
```
This is the result https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/publish-fedora-realtime-container-image/1442864157975121920 not sure if all is ok, there are some errors in the build log, @jordigilh please take a look

**Comment by dhiller** [comment]
/hold

waiting for https://github.com/kubevirt/kubevirtci/pull/690 to get merged first
